module/programs/validator/v1.js

```diff
@@ -31,18 +31,66 @@ module.exports = (req) => {
     addRolesInScope: function () {
       req.checkParams('_id').exists().withMessage('required program id');
       req.checkBody('roles').exists().withMessage('required program roles to be added');
+      req.checkBody('roles').isArray().withMessage('roles must be an array')
     },
     addEntitiesInScope: function () {
-      req.checkParams('_id').exists().withMessage('required program id');
-      req.checkBody('entities').exists().withMessage('required entities to be added');
+      req.checkParams('_id')
+				.exists()
+				.withMessage('required program id')
+				.isMongoId()
+				.withMessage('Invalid program ID')
+			req.checkBody('entities').exists().withMessage('required entities to be added')
+			const entities = req.body.entities
+			if (entities && typeof entities === 'object') {
+				for (const [key, value] of Object.entries(entities)) {
+					req.checkBody(`entities.${key}`).isArray().withMessage(`${key} should be an array`)
+				}
+			}
+      if (req.query.organizations === 'true') {
```

Comment:
@MallanagoudaB Please don't introduce any query param, hardcode for req body key